### PR TITLE
Refactoring to create design documents

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: www-src/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   },
   "scripts": {
     "postinstall": "./node_modules/.bin/brunch build",
-    "test": "/bin/true",
+    "test": "./node_modules/.bin/mocha test/unit/* --reporter spec --compilers coffee:coffee-script/register --colors",
```
Comment (nono) [inline www-src/package.json]: I think you can omit `./node_modules/.bin/`. This path is automatically added to `$PATH` when using `npm run`.

File: www-src/app/replicator/design_documents.coffee [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    @FilesAndFolderDesignDoc:
+        _id: "_design/#{@FILES_AND_FOLDER}"
+        version: 1
+        views:
+            "#{@FILES_AND_FOLDER}":
+                map: Object.toString.apply (doc) ->
+                    if doc.name?
+                        if doc.docType?.toLowerCase() is 'file'
+                            emit [doc.path, '2_' + doc.name.toLowerCase()]
+                        if doc.docType?.toLowerCase() is 'folder'
```
Comment (nono) [inline www-src/app/replicator/design_documents.coffee]: @kosssi you should probably add a comment for this view that explain _why_ it forces folders to come before files.

Comment (kosssi) [inline www-src/app/replicator/design_documents.coffee]: @nono I don't know ;) it's here https://github.com/cozy/cozy-mobile/pull/133/files#diff-0628999c8b29a3a2fa157635d0c095d1L32

File: www-src/app/replicator/design_documents.coffee [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                map: Object.toString.apply (doc) ->
+                    if doc.docType?.toLowerCase() is 'file'
+                        emit doc.binary?.file?.id
+
+
+    @PicturesDesignDoc:
+        _id: "_design/#{@PICTURES}"
+        version: 1
+        views:
+            "#{@PICTURES}":
```
Comment (nono) [inline www-src/app/replicator/design_documents.coffee]: @kosssi any idea why it is called pictures, even if its definition seems to have nothing specific to pictures?

File: www-src/app/replicator/replicator_backups.coffee [hunk 4]
```diff
@@ -124,15 +126,16 @@ module.exports =
         async.series [
             @ensureDeviceFolder.bind this
             ImagesBrowser.getImagesList
-            (callback) => @photosDB.query 'PhotosByLocalId', {}, callback
-            (cb) => @db.query 'FilesAndFolder',
+            (callback) =>
```
Comment (nono) [inline www-src/app/replicator/replicator_backups.coffee]: @kosssi you should rename `callback` in  `cb` to be consitent with the new function (and avoid confusion with the `callback` from the outer function syncPictures).

**Comment by nono** [comment]
Looks good to me

File: www-src/app/replicator/design_documents.coffee (hunk 3 above)
Comment (kosssi) [inline www-src/app/replicator/design_documents.coffee]: @nono I don't know too ^^ @jacquarg maybe can explain

**Comment by kosssi** [comment]
@nono me too

**Comment by nono** [comment]
Build is green, I merge. Thanks!

**Pull request merged**